invenio_rdm_records/services/review/service.py

```diff
@@ -123,12 +123,15 @@ def delete(self, identity, id_, revision_id=None, uow=None):
         if draft.parent.review.is_open:
             raise ReviewStateError(_("An open review cannot be deleted."))
 
-        # Delete request
-        current_requests_service.delete(
-            identity,
-            draft.parent.review.id,
-            uow=uow
-        )
+        # Delete request if request not in ['expired', 'declined']
+        # we keep the request in this case so the user can see the outcome of
+        # the request
+        if draft.parent.review.status not in ['expired', 'declined']:
```

Comment:
Minor: I think it's safer to use the ``is_closed/is_open`` properties here. You can only delete a request in ``created`` state (which is neither closed nor open):

```suggestion
        if not (draft.parent.review.is_closed or draft.parent.review.is_open):
```